nevergrad/optimization/recastlib.py

```diff
@@ -113,3 +115,171 @@ def __init__(self, *, method: str = "Nelder-Mead", random_restart: bool = False)
+        problem = self._create_pymoo_problem(self, objective_function)
+        pymoooptimize.minimize(problem, algorithm)
+        return None
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        """Reads messages from the thread in which the underlying optimization function is running
+        New messages are sent as "ask".
+        """
+        if self.num_objectives == 0:  # dummy ask i.e. not activating pymoo until num_objectives is set
+            data = self._rng.normal(
+                0, 1, self.dimension
+            )  # get a datapoint that is a random point in parameter space
+            candidate = self.parametrization.spawn_child().set_standardized_data(data)
+            return candidate
+        return super()._internal_ask_candidate()
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: float) -> None:
+        """Returns value for a point which was "asked"
+        (none asked point cannot be "tell")
+        """
+        x = candidate.get_standardized_data(
+            reference=self.parametrization
+        )  # dummy ask i.e. not activating pymoo until num_objectives is set
+        if self._messaging_thread is None:
+            return
+        if not self._messaging_thread.is_alive():  # optimizer is done
+            self._check_error()
+            return
+        messages = [m for m in self._messaging_thread.messages if m.meta.get("asked", False) and not m.done]
+        messages = [m for m in messages if m.meta["uid"] == candidate.uid]
+        if not messages:
+            raise RuntimeError(f"No message for evaluated point {x}: {self._messaging_thread.messages}")
+        messages[0].result = candidate.losses  # post all the losses, and the thread will deal with it
+
+    def _create_pymoo_problem(
+        self, optimizer: base.Optimizer, objective_function: tp.Callable[[tp.ArrayLike], float]
+    ):
+        # pylint:disable=import-outside-toplevel
+        from pymoo.model.problem import Problem  # type: ignore
+
+        class _PymooProblem(Problem):
+            def __init__(self, optimizer, objective_function):
+                self.objective_function = objective_function
+                super().__init__(
+                    n_var=optimizer.dimension,
+                    n_obj=optimizer.num_objectives,
+                    n_constr=0,  # constraints handled already by nevergrad
+                    xl=-math.pi * 0.5,
+                    xu=math.pi * 0.5,
+                    elementwise_evaluation=True,  # for 1-by-1 evaluation
+                )
+
+            def _evaluate(self, X, out, *args, **kwargs):
+                # pylint:disable=unused-argument
+                out["F"] = self.objective_function(
+                    np.tan(X)
+                )  # transform data so that it fits into bounds bounds
+
+        return _PymooProblem(optimizer, objective_function)
+
+
+class PymooOptimizer(base.ConfiguredOptimizer):
```

Comment:
How about simplifying a bit since everything is expected to be optimizers?

```suggestion
class Pymoo(base.ConfiguredOptimizer):
```

Reply:
Noted. Done.